packages/sanity/src/_internal/cli/commands/schema/extractSchemaCommand.ts

```diff
@@ -0,0 +1,31 @@
+import {type CliCommandDefinition} from '@sanity/cli'
+
+const description = 'Extracts a json representation of a sanity schema within a Studio context.'
+
+const helpText = `
+**Note**: This command is experimental and subject to change.
+
+Options
+  --workspace <name> The name of the workspace to generate a schema for
+  --path Optional path to specify destination of the schema file
+  --enforce-required-fields Makes the schema generated respect required fields in the schema. This can be used when generating types if you are only querying for published documents. Defaults to false.
```

Comment:
- This seems broader than just for querying - I would phrase this to be "treat fields marked as required as non-optional", as what it impacts is the resulting `optional` attributes
- "querying for published documents" doesn't really encompass this - both drafts and published documents can be incomplete/not follow schema (documents that were published prior to a field being introduced, for instance)

Reply:
> This seems broader than just for querying - I would phrase this to be "treat fields marked as required as non-optional", as what it impacts is the resulting optional attributes

True, and agree on your phrasing 👍 

>  This seems broader than just for querying - I would phrase this to be "treat fields marked as required as non-optional", as what it impacts is the resulting optional attributes
"querying for published documents" doesn't really encompass this - both drafts and published documents can be incomplete/not follow schema (documents that were published prior to a field being introduced, for instance).

Yeah, we do however consider the schema the source of truth so if anything diverges from your schema we don't give any guarantees with the types at least 🤔